File: oomtest/src/main/java/com/neoqee/oomtest/json/GsonUtil.java
package com.neoqee.oomtest.json;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;

public class GsonUtil {

    private static Gson gson = new Gson();

    public static <T> T fromJson(String json,Class<T> cls){
        return gson.fromJson(json,cls);
    }

    public static <T> T fromObject(Object object,Class<T> cls){
        String s = gson.toJson(object);
        return gson.fromJson(s,cls);
    }

}

File: skeleton/src/main/java/com/demo/skeleton/GsonUtil.kt
package com.demo.skeleton

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import java.lang.reflect.Type

object GsonUtil {
    private val gson=Gson()

    fun toJson(any: Any):String{
        return gson.toJson(any)
    }

    fun <T> fromJson(jsonStr:String,cls:Class<T>):T{
        return gson.fromJson(jsonStr,cls)
    }

    fun <T> fromJson(jsonStr: String,type:Type):T{
//        var myType=object : TypeToken<Any>(){}.type
        return gson.fromJson(jsonStr,type)
    }

    fun <T> fromObject(any: Any,type: Type):T{
        val json = toJson(any)
        return fromJson(json,type)
    }

}